src/UnderlineNav/UnderlineNavLink.tsx

```diff
@@ -0,0 +1,97 @@
+import React, {forwardRef, useLayoutEffect, useRef, useContext, MutableRefObject} from 'react'
+import Box from '../Box'
+import {merge, SxProp} from '../sx'
+import {get} from '../constants'
+import {IconProps} from '@primer/octicons-react'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '@radix-ui/react-polymorphic'
+import {UnderlineNavContext} from './UnderlineNavContext'
+
+// adopted from React.AnchorHTMLAttributes
+type LinkProps = {
+  download?: string
+  href?: string
+  hrefLang?: string
+  media?: string
+  ping?: string
+  rel?: string
+  target?: string
+  type?: string
+  referrerPolicy?: React.AnchorHTMLAttributes<HTMLAnchorElement>['referrerPolicy']
+}
+
+export type UnderlineNavLinkProps = {
+  /**
+   * Primary content for an NavLink
+   */
+  children?: React.ReactNode
+  /**
+   * Callback that will trigger both on click selection and keyboard selection.
+   */
+  onSelect?: (event: React.MouseEvent<HTMLLIElement> | React.KeyboardEvent<HTMLLIElement>) => void
+  /**
+   * Is the `Link` is currently selected?
+   */
+  selected?: boolean
```

Comment:
Should accept an `aria-current` prop instead of `selected`? That would be more consistent with NavList: https://primer.style/react/NavList#navlistitem